server/mscalendar/oauth2.go

```diff
@@ -71,7 +72,13 @@ func (app *oauth2App) CompleteOAuth2(authedUserID, code, state string) error {
 		return err
 	}
 
-	client := app.Remote.MakeClient(ctx, tok, app.Store, mattermostUserID, store.GetCheckUserStatus(app.Store, app.Logger, mattermostUserID), store.GetChangeUserStatus(app.Store, app.Logger, mattermostUserID, app.Poster))
+	tokenHelpers := &serializer.UserTokenHelpers{
+		CheckUserStatus:      app.Store.MakeCheckUserStatus,
+		ChangeUserStatus:     app.Store.MakeChangeUserStatus,
+		RefreshAndStoreToken: app.Store.RefreshAndStoreToken,
+	}
+
+	client := app.Remote.MakeClient(ctx, tok, mattermostUserID, app.Poster, tokenHelpers)
```

Comment:
I guess this gets run during the Oauth, so I thought this won't need the refresh token part but is changing that now.

Reply:
> so I thought this won't need the refresh token part here.

I would say that we should still call `MakeUserClient`, since this is a client for a user, and not a superuser client